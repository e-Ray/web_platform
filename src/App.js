import React, { Component } from 'react';
import Routes from './environment/router/routes';

class App extends Component {
  render() {
    return (
      <Routes />
    );
  }
}

export default App;
