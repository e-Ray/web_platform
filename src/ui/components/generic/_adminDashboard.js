import React, { Component } from 'react';
import { observer } from 'mobx-react';
import { observable, autorun } from 'mobx';
import { Paper } from 'material-ui/Paper'
import { List, ListItem } from 'material-ui/List';
import { UserCard, ErayCard } from './';
import { ref } from '../../../api/Auth/_constants';

@observer
class AdminDashboard extends ComponentÂ {

	@observable users = [];
	@observable erays = [];
	@observable erayItems = [];
	@observable items = [];
	@observable counter = 0;
	@observable user = "";
	@observable eray = "";

	constructor(props){
		super(props);
		ref.child('/users/').on('value', (snapshot)=>{

			snapshot.forEach((userSnapshot) => {

				let lastname = "";
				let firstname = "";
				let email = "";
				let hash = userSnapshot.key;
				ref.child('users/'+hash+'/info/')
					.on('value', (infoSnapshot) =>{
						lastname = infoSnapshot.val().lastname;
						firstname = infoSnapshot.val().firstname;
						email = infoSnapshot.val().email;
					
					});

				this.users.push({
					lastname: lastname,
					firstname: firstname,
					email: email,
					hash: hash
				});
			}

			);

		});
		ref.child('/erays/eraylist').on('value', (snapshot)=>{

			snapshot.forEach((eraySnapshot) => {
				let owner = eraySnapshot.val().owner;
				let location = eraySnapshot.val().location;
				let commissioning = eraySnapshot.val().commissioning;
				let id = eraySnapshot.key;
				this.erays.push({
					owner: owner,
					location: location,
					commissioning: commissioning,
					id: id
				});
			}

			);

		});
		autorun(()=>{
			this.items = this.users.slice().map((item) =>
					<div key={item.hash + Math.random()}>
						<ListItem
							key={item.hash + Math.random()}
							primaryText={item.lastname + ", " + item.firstname}
							onTouchTap={() => {this.user = item;}}/>
					</div>
			);
			this.erayItems = this.erays.slice().map((eray) =>
					<div key={eray.id + Math.random()}>
						<ListItem
							key={eray.id + Math.random()}
							primaryText={eray.location + ", " + eray.id}
							onTouchTap={() => {this.eray = eray;}}/>
					</div>
			);

		});
		this.clearArrays = this.clearArrays.bind(this);


	}

	clearArrays() {
		this.items = [];
		this.users = [];
		this.erayItems = [];
		this.erays = [];
	}

	getUserCard() {
		if(this.user !== ""){
			return <div><UserCard user={this.user} clearArrays={this.clearArrays} key={this.user.hash}/></div>;
		}
	}

	getErayCard() {
		if(this.eray !== ""){
			return <div><ErayCard eray={this.eray} clearArrays={this.clearArrays} key={this.eray.id} /></div>;
		}
	}

	render(){
		
		return (
			<div>
			<div id="row">
			<div id="col-2-left">
				<List style={{maxHeight: 250, overflow: 'auto'}}>
					{this.items}
				</List>

			</div>
			<div id="col-2-left">
				<List style={{maxHeight: 250, overflow: 'auto'}}>
					{this.erayItems}
				</List>
			</div>
			</div>
			<div id="row">
				<div id="col-2-left">
				{this.getUserCard()}
			</div>
				<div id="col-2-left">
				{this.getErayCard()}
				</div>
			</div>
			</div>
		);

	}
}

export default AdminDashboard;
