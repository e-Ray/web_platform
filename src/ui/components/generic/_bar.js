import React from 'react';
import AppBar from 'material-ui/AppBar';
import { Link } from 'react-router';
import Drawer from 'material-ui/Drawer';
import MenuItem from 'material-ui/MenuItem';
import { logout } from '../../../api/Auth/_auth';
import FlatButton from 'material-ui/FlatButton';
import { firebaseAuth } from '../../../api/Auth/_constants';
import Divider from 'material-ui/Divider';

/**
 * A simple example of `AppBar` with an icon on the right.
 * By default, the left icon is a navigation-menu.
 */
export default class AppBarMenu extends React.Component {

	constructor(props) {
    super(props);
    this.state = {open: false};

    this.handleLogout = this.handleLogout.bind(this);
  	}

  	handleToggle = () => this.setState({open: !this.state.open});

  	handleClose = () => this.setState({open: false});

    handleLogout = () => {
      logout();

      console.log('logged out');
        }
    componentDidMount(){
      firebaseAuth().onAuthStateChanged(user => {
        if(!user){
          window.location.reload();
        }
      });
    }

	render(){

		return(
			<div>

      	<AppBar
          title="e.Ray"
          onLeftIconButtonTouchTap={this.handleToggle}
          iconElementRight={<FlatButton label="Logout" onTouchTap={this.handleLogout} href="/notloggedin" />}
        	/>

       	 <Drawer
          docked={false}
          width={200}
          open={this.state.open}
          onRequestChange={(open) => this.setState({open})}
        >

          <MenuItem onTouchTap={this.handleClose} containerElement={<Link to="Dashboard" />}>Dashboard</MenuItem>
					<Divider />
					<MenuItem onTouchTap={this.handleClose} containerElement={<Link to="DetailPagePerf" />}>Leistung</MenuItem>
					<MenuItem onTouchTap={this.handleClose} containerElement={<Link to="DetailPageWL" />}>Wasserstand</MenuItem>
					<MenuItem onTouchTap={this.handleClose} containerElement={<Link to="DetailPageRpm" />}>RPM</MenuItem>
			    <MenuItem onTouchTap={this.handleClose} containerElement={<Link to="DetailPageWaterTemp" />}>Wassertemperatur</MenuItem>
					<MenuItem onTouchTap={this.handleClose} containerElement={<Link to="DetailPageTemp" />}>Lufttemperatur</MenuItem>
					<MenuItem onTouchTap={this.handleClose} containerElement={<Link to="DetailPageRain" />}>Niederschlagsmenge</MenuItem>
					<MenuItem onTouchTap={this.handleClose} containerElement={<Link to="DetailPageWindSpeed" />}>Windgeschwindigkeit</MenuItem>
					<MenuItem onTouchTap={this.handleClose} containerElement={<Link to="DetailPageWindDir" />}>Windrichtung</MenuItem>
					<Divider />
					<MenuItem onTouchTap={this.handleClose} containerElement={<Link to="InfoPage" />}>Information</MenuItem>
					<Divider />
					<MenuItem onTouchTap={this.handleClose} containerElement={<Link to="HelpPage" />}>Hilfe-Seite</MenuItem>
        </Drawer>

  		</div>
		);
	}

}
