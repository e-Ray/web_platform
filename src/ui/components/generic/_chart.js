import React, { Component } from 'react';
import { Line } from 'react-chartjs-2';
import { ref } from '../../../api/Auth/_constants';
import { observer } from 'mobx-react';
import Loader from 'react-loader';
import { observable, action } from 'mobx';
import moment from 'moment';


@observer
class Chart extends Component {
  @observable daten = [];
  @observable labels = [];
  @observable daysSeen = 0;
  
  constructor(props) {
    super(props);
    this.state={
      range: this.props.range,
      sensor: this.props.sensor
    };


  }
  componentDidMount(){
    this.getData();
  }
  componentDidUpdate(prevProps, prevState){
    if(prevProps.range !== this.props.range){
      this.daysSeen=0;
      this.getData();
    }
  }

@action
  getData(){
    this.daysSeen = 0;
    this.daten.clear();
    this.labels.clear();
    
    let range = this.props.range;
    let iterator = new Date();
    iterator.setDate(this.props.date.getDate()-range+1);

    while(range > 0){

      ref.child('/erays/'+this.props.eray+'/'+this.props.sensor+moment(iterator).format("/YYYY_M_D/"))
        .once('value',(daySnapshot) =>{

          // more than 6 Days -> average values
          if(this.props.range >= 7){
          let values = [];
          let label = '';

          daySnapshot.forEach((valueSnapshot) =>{
              values.push(valueSnapshot.val().value);
              let date = valueSnapshot.val().date.replace(/_/g,"/");
              label = moment(date).format("M/D/YYYY");
          });

          let total = 0;
          for (let i = 0; i<values.length; i++){
              total += values[i];
          }

          this.daten.push((total/values.length));
          this.labels.push(label);
          } 
          // less than 7 days -> all values
          else {
            daySnapshot.forEach((valueSnapshot) =>{
              this.daten.push(valueSnapshot.val().value);
              let date = valueSnapshot.val().date.replace(/_/g,"/");
              let time = valueSnapshot.val().timestamp.replace("_",":");
              this.labels.push(moment(date).format("M/D/YYYY")+'   '+time);

          });
          }

          this.daysSeen++;
         
      });
      iterator = moment(iterator).add(1, 'days');
      range--;
      this.daysSeen++;
    };

  }



	render() {

    if (this.daysSeen === ((this.props.range)*2)){
		return(

       <div id="col-1">
				   <Line redraw data={ {
            labels: this.labels.slice(),
            datasets: [
              {
                label: this.props.sensor,
                fill: true,
                lineTension: 0.1,
                backgroundColor: 'rgba(75,192,192,0.4)',
                borderColor: 'rgba(75,192,192,1)',
                borderCapStyle: 'butt',
                borderDash: [],
                borderDashOffset: 0.0,
                borderJoinStyle: 'miter',
                pointBorderColor: 'rgba(75,192,192,1)',
                pointBackgroundColor: '#fff',
                pointBorderWidth: 1,
                pointHoverRadius: 5,
                pointHoverBackgroundColor: 'rgba(75,192,192,1)',
                pointHoverBorderColor: 'rgba(220,220,220,1)',
                pointHoverBorderWidth: 2,
                pointRadius: 1,
                pointHitRadius: 10,
                data: this.daten.slice()
              }
            ]
          }} 
          width={ this.props.width } height={ this.props.height } key={Math.random()}
            options={ { maintainAspectRatio: false, responsive: true, legend: { display: false, }, yAxes: [{label: "Label"}] } } />
           
        </div>
		);


    }

    if(this.props.mode === "dashboard"){
      return <h6>Loading ...</h6>
    }
    return <div><Loader loaded={false}/></div>

	}
}


export default Chart;
