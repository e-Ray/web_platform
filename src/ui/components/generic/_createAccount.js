import React, { Component } from 'react'
import { auth } from '../../../api/Auth/_auth'
import Dialog from 'material-ui/Dialog';
import TextField from 'material-ui/TextField';
import { observer } from 'mobx-react';
import { observable } from 'mobx';
import RaisedButton from 'material-ui/RaisedButton';

@observer
class CreateAccount extends Component {

  @observable data = {
    lastname: "",
    firstname: "",
    email: "",
    password: "",
    passwordVerification: "",
  };

  constructor(props){
    super(props);
    this.state = {
      verificationError: "",
      emailError: "",
      passwordError: ""
    };

    this.onLastnameChangeHandler = this.onLastnameChangeHandler.bind(this);
    this.onFirstnameChangeHandler = this.onFirstnameChangeHandler.bind(this);
    this.onEmailChangeHandler = this.onEmailChangeHandler.bind(this);
    this.onPasswordChangeHandler = this.onPasswordChangeHandler.bind(this);
    this.onPasswordVerificationHandler = this.onPasswordVerificationHandler.bind(this);
  }

  handleSubmit = (e) => {
    this.resetError();
    e.preventDefault();
    if(this.data.password === this.data.passwordVerification){
      auth(this.data.email, this.data.password, this.data.firstname, this.data.lastname).catch((error) => this.handleError(error.code));
      console.log("auth: " + this.data.email + " " + this.data.password);
    }else{
      this.setState({verificationError: "Passwords do not match!"});
    }
  }

  handleError(errorCode){
    if(errorCode === "auth/invalid-email"){
      this.setState({emailError: "The E-Mail format is invalid!"});
    }else if(errorCode === "auth/email-already-in-use"){
      this.setState({emailError: "The E-Mail is already in use!"});
    }else if(errorCode === "auth/weak-password"){
      this.setState({passwordError: "The password is not strong enough!"});
    }
  }

  resetError(){
    this.setState({ verificationError: "" });
    this.setState({ emailError: "" });
    this.setState({ passwordError: "" });
  }

  onLastnameChangeHandler(e){
    this.data.lastname = e.target.value;
  }

  onFirstnameChangeHandler(e){
    this.data.firstname = e.target.value;
  }

  onEmailChangeHandler(e){
    this.data.email = e.target.value;
  }

  onPasswordChangeHandler(e){
    this.data.password = e.target.value;
  }

  onPasswordVerificationHandler(e){
    this.data.passwordVerification = e.target.value;
  }

  render () {
    return (
      <div>
        <Dialog
            title="Registration"

            modal={false}
            open={this.props.open}
            onRequestClose={this.props.close}
            >
            <TextField hintText="Last Name"
                       value={this.data.lastname}
                       onChange={this.onLastnameChangeHandler}/>
            <TextField hintText="First Name"
                       value={this.data.firstname}
                       onChange={this.onFirstnameChangeHandler}/>
            <TextField hintText="E-Mail"
                       value={this.data.email}
                       errorText={this.state.emailError}
                       onChange={this.onEmailChangeHandler}/>
            <TextField hintText="Password"
                       type="password"
                       value={this.data.password}
                       errorText={this.state.passwordError}
                       onChange={this.onPasswordChangeHandler}/>
            <TextField hintText="Password Verification"
                       type="password"
                       errorText={this.state.verificationError}
                       value={this.data.passwordVerification}
                       onChange={this.onPasswordVerificationHandler}/>
            <RaisedButton type="submit" label="Create" primary={true} onClick={this.handleSubmit}/>
        </Dialog>
      </div>

    )
  }
}
export default CreateAccount;
