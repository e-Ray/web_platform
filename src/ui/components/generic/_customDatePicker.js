import React from 'react';
import DatePicker from 'material-ui/DatePicker';
import RaisedButton from 'material-ui/RaisedButton';

const CustomDatePicker = ({ dayTo, dayFrom, handler }) => {
  let bool = false;

  return (
    <div>
      <div id="col-5">
        <DatePicker
          hintText="From" autoOk
          onChange={(d, value) => { dayFrom = new Date(value.getTime()); bool = true; }}
        />
      </div>
      <div id="col-5">
        <DatePicker
          hintText="To" autoOk
          onChange={(d, value) => { dayTo = new Date(value.getTime()); bool = true; }}
        />
      </div>
      <div id="col-5">
        <RaisedButton
          label="ok" onClick={() => {
            if (bool) {
              if (dayTo.getTime() > dayFrom.getTime()) {
                handler(dayTo, dayFrom); bool = false;
              } else {
                handler(dayFrom, dayTo); bool = false;
              }
            }
          }}
        />
      </div>
    </div>);
};
CustomDatePicker.propTypes = {
  dayTo: React.PropTypes.instanceOf(Date),
  dayFrom: React.PropTypes.instanceOf(Date),
  handler: React.PropTypes.func,
};
export default CustomDatePicker;
