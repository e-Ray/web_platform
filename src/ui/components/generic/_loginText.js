import React from 'react';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import { Link } from 'react-router'; 
import { login, logout } from '../../../api/Auth/_auth';
import Dialog from 'material-ui/Dialog';




class LoginText extends React.Component {

  constructor(props){
   super(props);
   this.state={
      Username: '',
      Password: '',
      loading:true,
      userError: '',
      passwordError: '',
   };

    this.onNameChangeHandler = this.onNameChangeHandler.bind(this);
    this.onPasswordChangeHandler = this.onPasswordChangeHandler.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleLogout = this.handleLogout.bind(this);
    this.handleError = this.handleError.bind(this);
    }

  onNameChangeHandler(e){
    this.setState({Username: e.target.value});
  }

  onPasswordChangeHandler(e){
    this.setState({Password: e.target.value});
  }
  

  handleSubmit = (e) => {
    
    e.preventDefault();
    login(this.state.Username, this.state.Password).catch(
      (error)=>{this.handleError(error)});
    
  }

  handleError(error){
    var errorCode = error.code;

      if(errorCode === 'auth/user-not-found')
         this.setState({userError: 'Account wurde nicht gefunden'});
      else if(errorCode === 'auth/user-disabled')
          this.setState({userError: 'Account wurde stillgelegt'});
      else if(errorCode === 'auth/invalid-email')
          this.setState({userError: 'Ungültige E-Mail Adresse'});
      else if(errorCode === 'auth/wrong-password')
          this.setState({passwordError: 'Ungültiges Passwort'});
  }
  handleLogout = (e) => {
    logout();
    console.log('logout');
  }


  render() {
    return (
      <div> 

      <Dialog
           title="Login"
           modal={true}
           open={true}
          >   
  
        <TextField  hintText="Username" 
                    errorText={this.state.userError}
                    value={this.state.Username} 
                    onChange={this.onNameChangeHandler}/>
        <br></br>
        
        <TextField  hintText="Password" 
                    errorText={this.state.passwordError}
                    type="password"
                    value={this.state.Password} 
                    onChange={this.onPasswordChangeHandler}/>


        <RaisedButton label="Login" onClick={this.handleSubmit}/>
        <br></br>
        <RaisedButton label="Create Account" 
                      containerElement={<Link to="/dashboard" />}/>
        <br></br>
        <RaisedButton label="Log Out" onClick={this.handleLogout}/>
        <h1>Username: {this.state.Username}</h1>
        <h2>Password: {this.state.Password}</h2>
      </Dialog>
      
      </div>
      )
  }

}

export default LoginText;
