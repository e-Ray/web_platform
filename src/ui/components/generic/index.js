export { default as DetailPage } from './_detailPage';
