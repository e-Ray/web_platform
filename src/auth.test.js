import React from 'react';
import ReactDOM from 'react-dom';
import App from './ui/pages/index';
import LoginText from './ui/components/generic/_loginText';
import renderer from 'react-test-renderer';


describe('Login ', () => {
  it('with bad E-Mail', () => {
    require('./ui/components/generic/_loginText').default;
    const component = renderer.create(
      <LoginText testing userName="bad E-Mail" />,
  );


    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

  // manually trigger the callback
    tree.props.handleSubmitForTests;

  // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });


  it('with wrong E-Mail', () => {
    const component = renderer.create(
      <LoginText testing userName="bla@bla.bla" />,
  );


    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

  // manually trigger the callback
    tree.props.handleSubmit;
  // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('with wrong password', () => {
    const component = renderer.create(
      <LoginText testing userName="test123@test.de" password="d" />,
  );


  // manually trigger the callback
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

  // manually trigger the callback
    tree.props.handleSubmit;
  // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('with good E-Mail and good password', () => {
    const component = renderer.create(
      <LoginText testing userName="test123@test.de" password="123456" />,
  );


    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

  // manually trigger the callback
    tree.props.handleSubmit;
  // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
