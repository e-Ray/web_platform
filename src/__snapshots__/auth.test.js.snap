exports[`Login  with bad E-Mail 1`] = `
<a
  password={undefined}
  passwordError=""
  userError=""
  userName="bad E-Mail" />
`;

exports[`Login  with bad E-Mail 2`] = `
<a
  password={undefined}
  passwordError=""
  userError=""
  userName="bad E-Mail" />
`;

exports[`Login  with good E-Mail and good password 1`] = `
<a
  password={undefined}
  passwordError=""
  userError=""
  userName="test123@test.de" />
`;

exports[`Login  with good E-Mail and good password 2`] = `
<a
  password={undefined}
  passwordError=""
  userError=""
  userName="test123@test.de" />
`;

exports[`Login  with wrong E-Mail 1`] = `
<a
  password={undefined}
  passwordError=""
  userError=""
  userName="bla@bla.bla" />
`;

exports[`Login  with wrong E-Mail 2`] = `
<a
  password={undefined}
  passwordError=""
  userError=""
  userName="bla@bla.bla" />
`;

exports[`Login  with wrong password 1`] = `
<a
  password={undefined}
  passwordError=""
  userError=""
  userName="test123@test.de" />
`;

exports[`Login  with wrong password 2`] = `
<a
  password={undefined}
  passwordError=""
  userError=""
  userName="test123@test.de" />
`;
